corehq/apps/reports/management/commands/prepare_projects_for_case_list_explorer.py

```diff
@@ -0,0 +1,111 @@
+from datetime import datetime
+
+from django.core.management import BaseCommand
+
+from dimagi.utils.couch.database import iter_docs
+from corehq.apps.case_search.models import DomainsNotInCaseSearchIndex
+from corehq.apps.domain.models import Domain
+from corehq.apps.es import CaseES, CaseSearchES
+from corehq.pillows.case_search import CaseSearchReindexerFactory, domain_needs_search_index
+
+
+class Command(BaseCommand):
+    """Helper tool for preparing legacy projects on CommCare HQ instances for the General Release
+    of Case List Explorer. This is done by ensuring data is synced between the Elasticsearch CaseIndex
+    and CaseSearchIndex.
+    """
+
+    def add_arguments(self, parser):
+        parser.add_argument(
+            '--project',
+            action='store',
+            dest='domain',
+            help='Specify a specific project to run data a integrity check and re-sync on.'
+        )
+
+    def handle(self, **options):
+        domain = options.pop('domain')
+        if domain is not None:
+            if DomainsNotInCaseSearchIndex.objects.count() > 0:
+                self.stdout.write("Please finish running the complete initial sync by "
+                                  "running this management command without the --project "
+                                  "argument.")
+                return
+            self.run_domain_re_sync(domain)
+            return
+
+        self.migrate_domains_not_in_case_search_index()
+        self.stdout.write("Initial sync complete.")
+        self.run_data_integrity_checks()
+
+    def migrate_domains_not_in_case_search_index(self):
+        count = DomainsNotInCaseSearchIndex.objects.count()
+        if count > 0:
+            self.stdout.write(f"\n\nFound {count} project(s) to sync...")
+
+            confirm = input("Please make sure you run this command inside a terminal multiplexer "
+                            "like tmux or screen as this could take some time. Ready to continue? [y/n] ")
+            if confirm.lower() != 'y':
+                self.stdout.write("Aborting initial sync...")
+                return
+        else:
+            return
+
+        # delete DomainsNotInCaseSearchIndex for zero-case domains
+        DomainsNotInCaseSearchIndex.objects.filter(estimated_size__lte=0).all().delete()
+
+        for tracker in DomainsNotInCaseSearchIndex.objects.order_by('estimated_size').all():
```

Comment:
Any specific reason to order by estimated_size? Especially because it appears that the `estimated_size` field is not indexed? Does ordering matter?

Reply:
we want to do the biggest projects first...